docs/odh/fetch-upstream-release.md

```diff
@@ -0,0 +1,430 @@
+This repository is a fork of [kserve/kserve](https://github.com/kserve/kserve)
+repository to integrate with Open Data Hub platform. The releases on ODH
+platform have a faster cadence than the upstream repository. Still, an upstream
+stable release is used as a base for any releases of the ODH Project. We
+accomplish this by creating a branch in ODH fork based from an upstream release
+tag. Any integration changes to ODH Platform are added on top of the created
+branch.
+
+This guide outlines the process to fetch a stable upstream release to ODH fork.
+
+# Prerequisites
+
+As you will be are creating a _new_ branch, the configured branch protections
+may not apply to it. Despite this, you must still adhere to [GitHub
+flow](https://docs.github.com/en/get-started/using-github/github-flow).
+
+You will need:
+
+* Push rights to
+  [opendatahub-io/kserve](https://github.com/opendatahub-io/kserve/) repository.
+  * This is to allow you doing the initial push of the release branch.
+  * If you don't have push rights, you can ask somebody else to do the initial
+    push.
+* Your own fork of
+  [opendatahub-io/kserve](https://github.com/opendatahub-io/kserve/).
+  * This is to stay adhered to [GitHub
+    flow](https://docs.github.com/en/get-started/using-github/github-flow).
+  * You can create a fork on your account using this link:
+    https://github.com/opendatahub-io/kserve/fork.
+  * If you have push rights to
+    [opendatahub-io/kserve](https://github.com/opendatahub-io/kserve/) the fork
+    may be perceived as optional. Still, we prefer to keep the repository clean
+    from working branches and open pull requests from contributor forks.
+* Your own fork of
```

Comment:
Pushing a new branch would not require you to have a fork of odh, just if you want to contribute with it, right?

Reply:
> Pushing a new branch would not require you to have a fork of odh, just if you want to contribute with it, right?

This is right. But you may require to send a PR with the adaptations. So, you may need your own fork.

> I see you are describing below, how to configure your dev environment, wouldn't it be better to add those steps in the CONTRIBUTING_GUIDE.md?

Hmm... it can make sense if it is a pattern that we want all ODH contributors to following. Should I move it?

Reply:
I think that makes sense and might be the easier way to handle three different forks for the same project.

Reply:
That section is now moved to the contributing file.